pkg/middleware/stored_session.go

```diff
@@ -21,16 +22,7 @@ type StoredSessionLoaderOptions struct {
 	// Session storage backend
 	SessionStore sessionsapi.SessionStore
 
-	// How often should sessions be refreshed
-	RefreshPeriod time.Duration
-
-	// Provider based session refreshing
-	RefreshSession func(context.Context, *sessionsapi.SessionState) (bool, error)
-
-	// Provider based session validation.
-	// If the sesssion is older than `RefreshPeriod` but the provider doesn't
-	// refresh it, we must re-validate using this validation.
-	ValidateSession func(context.Context, *sessionsapi.SessionState) bool
+	Options *options.Options
```

Comment:
We are trying to make the parts of this project smaller and more well defined. In general, we shouldn't be passing the entire options structure around like this, but only the parts that are actually needed.

What is new that is needed in the options? Can that be passed in instead?

Reply:
If i remember correctly - originally the provider was defined/process during the initial middleware was called.

`		RefreshSession:  opts.GetProvider().RefreshSession,
		ValidateSession: opts.GetProvider().ValidateSession,`

and was problematic as we haven't yet actually checked the session for which provider is actually required.

The only way at the time without completely rewriting the whole middleware was to pass the options struct in, and then delay this function until we know what provider we actually need.

I wasn't really happy that I had to do this either, but I figured that as long as I am passing the pointer that it is "okish"